packages/connector-aliyun-dm/src/utils.ts

```diff
@@ -0,0 +1,87 @@
+import { createHmac } from 'crypto';
+
+import { has } from '@silverhand/essentials';
+import got from 'got';
+
+export type { Response } from 'got';
+export type SendEmailResponse = { EnvId: string; RequestId: string };
+
+// Aliyun has special escape rules.
+// https://help.aliyun.com/document_detail/29442.html
+const escaper = (string_: string) =>
+  encodeURIComponent(string_)
+    .replace(/\*/g, '%2A')
+    .replace(/'/g, '%27')
+    .replace(/!/g, '%21')
+    .replace(/"/g, '%22')
+    .replace(/\(/g, '%28')
+    .replace(/\)/g, '%29')
+    .replace(/\+/, '%2B');
+
+export const getSignature = (
+  parameters: Record<string, string>,
+  secret: string,
+  method: string
+) => {
+  const canonicalizedQuery = Object.keys(parameters)
+    .slice()
+    .sort()
```

Comment:
nit: sort after the filter?

Reply:
Feels like the two different places of placing `.sort()` are the same. Have no idea why the lint throws this error.
<img width="1034" alt="image" src="https://user-images.githubusercontent.com/15182327/165211688-cca71962-5425-43ec-b4dd-9ac6f632370b.png">

Reply:
you are talking about 2 issues:
- sort position
- sort is mutating